File: common/src/main/java/dev/number6/comprehend/results/SentimentResultsToMessageSentimentTotals.java
package dev.number6.comprehend.results;

import com.amazonaws.services.comprehend.model.DetectSentimentResult;

import java.util.Collection;
import java.util.HashMap;
import java.util.Map;
import java.util.function.Function;

public class SentimentResultsToMessageSentimentTotals implements Function<Collection<DetectSentimentResult>, Map<String, Integer>> {

    @Override
    public Map<String, Integer> apply(Collection<DetectSentimentResult> detectSentimentResults) {
        Map<String, Integer> sentimentTotals = new HashMap<>();
        detectSentimentResults.forEach(s -> sentimentTotals.merge(s.getSentiment(), 1, (i1, i2) -> i1 + i2));
        return sentimentTotals;
    }
}

File: common/src/main/java/dev/number6/comprehend/results/SentimentResultsToMessageSentimentTotals.kt
package dev.number6.comprehend.results

import com.amazonaws.services.comprehend.model.DetectSentimentResult
import java.util.*
import java.util.function.Consumer
import java.util.function.Function

class SentimentResultsToMessageSentimentTotals : Function<Collection<DetectSentimentResult>, Map<String, Int>> {
    override fun apply(detectSentimentResults: Collection<DetectSentimentResult>): Map<String, Int> {
        val sentimentTotals: MutableMap<String, Int> = HashMap()
        detectSentimentResults!!.forEach(Consumer { s: DetectSentimentResult? -> sentimentTotals.merge(s!!.sentiment, 1) { i1: Int, i2: Int -> i1 + i2 } })
        return sentimentTotals
    }
}